### One Tailed (Right Tailed) - One-Sample Z-Test

**1 - What is it?**

A. `Right One-Tailed One-Sample Z-Test` check the means of sample is significantly higher from the population mean (when population standard devidation is known and smaple size is large, usually n>=30).

`Think of i like` "Is my small group's average weight `really higher (significantly)` from the entire country's average weight, or is it just random chance"

**2 - Rea-Life Example - Right-Tailed One-Sample Z-Test**

Scenario: Your marketing team says : `Our new landing page increases average session time beyond 5 minutes`

You want to test if it's really greater than 5 minutes, not just different - So it's a one-tailed test (specifically, right-tailed).

**3 - Hypotheses**

- H0 (Null Hypotheses): population mean (meau) = 5 (no improvement)
- H1 (Alternate Hypotheses): sample mean > population mean (meau) (mean time increased)

This is a **right-tailed** test because we're checking "greater than."

**4 - Formula**

Z = (sample mean - population mean) / (population standard deviation/sqrt(sample size))

Z = (x bar - μ) / (σ/sqrt(n))

**Where: / Parameters**
- population mean (meau) : 5 min
- population standard deviation (σ) : 1 min
- sample mean (x bar) : 5.5 min
- sample size (n) : 50
- significance level (a) : 0.05

**4 - Formula**

Z = (sample mean - population mean)/(population standard deviation/sqrt(sample size)​)

Z = (x bar - μ) / (σ/sqrt(n))

**Where:**
- x bar  = sample mean
- μ (meu) = population mean
- σ (sigma) = population standard deviation
- n = sample size

**Python Implementation (Step-by-Step)**

In [7]:
import numpy as np
from scipy import stats

# ------------------------------------
# Step 1: Define parameters
# ------------------------------------
population_mean = 5 # average time on site (in minutes)
population_std = 1  # σ = population standard deviation
sample_mean = 5.5   # x̄ = sample mean 
sample_size = 50    # n = number of visitors in sample
alpha = 0.05        # significance level

# -------------------------------------
# Step 2: Calculate Standard Error (SE)
# -------------------------------------
standard_error = population_std / np.sqrt(sample_size)
print(f"Standard Error (SE): {round(standard_error,4)}")

# --------------------------------------
# Step 3: Calculate Z-Statistic
# --------------------------------------
z_stat = (sample_mean - population_mean) / standard_error
print(f"Z-Statistic: {round(z_stat,4)}")

# ---------------------------------------
# Step 4: Calculate one-tailed p-value (right tailed test)
# ----------------------------------------
p_value = 1 - stats.norm.cdf(z_stat)
print(f"P-Value: {round(p_value,4)}")

# -----------------------------------------
# Step 5: Decision
# -----------------------------------------
if p_value < alpha:
    print("Reject H0: The landing page signficantly increased session time.")
else:
    print("Fail to Reject H0: No signficant increase in session tiem.")

Standard Error (SE): 0.1414
Z-Statistic: 3.5355
P-Value: 0.0002
Reject H0: The landing page signficantly increased session time.


**6 - Business Interpretation**

- Z = 3.53 -> sample mean is 3.5 standard errors above the population mean.
- p = 0.0002 -> if the old landing page was truly average, the chance of seeing such a high sample mean (or higher) is 0.02%!
- So, your new page **really did** increase time - not by random luck.